src/XMakeCommandLine/XMake.cs

```diff
@@ -1363,6 +1357,30 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
         }
 
         /// <summary>
+        /// If on Unix, check if the string looks like a file path.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// </summary>
+        private static bool LooksLikeUnixFilePath(string value)
+        {
+#if MONO
+            var firstSlash = value.IndexOf('/');
```

Comment:
I agree -- this code matches the bottom of the MaybeAdjustFilePath in FileUtilities. It would be nice to have a common function. Also, I've been trying to convert ifdefs to runtime checks. Please replace the ifdef with a if(NativeMethodsShared.IsUnixLike).